imagebuildah/stage_executor.go

```diff
@@ -1214,7 +1214,7 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 				}
 			}
 		} else {
-			if s.executor.squash {
+			if s.executor.squash || !squashToggle {
```

Comment:
A stage executor method should _not_ be modifying the executor that owns it, since that executor can be executing multiple stages at the same time.

Reply:
Yes simplified the logic.